SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h

```diff
@@ -7,6 +7,165 @@
+
+	UPROPERTY(EditAnywhere, config, meta = (EditCondition = "!bAllPermissions", ConfigRestartRequired = false, DisplayName = "Component queries"))
+	TArray<FString> Components;
+};
+
+USTRUCT()
+struct FLoginRateLimitSection
+{
+	GENERATED_BODY()
+
+	FLoginRateLimitSection()
+		: Duration()
+		, RequestsPerDuration(0)
+	{
+
+	}
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Duration"))
+	FString Duration;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Requests per duration", ClampMin = "1", UIMin = "1"))
+	int32 RequestsPerDuration;
+};
+
+USTRUCT()
+struct FWorkerTypeLaunchSection
+{
+	GENERATED_BODY()
+
+	FWorkerTypeLaunchSection()
+		: WorkerTypeName()
+		, WorkerPermissions()
+		, MaxConnectionCapacityLimit(0)
+		, bLoginRateLimitEnabled(false)
+		, LoginRateLimit()
+		, Columns(1)
+		, Rows(1)
+		, ManualWorkerConnectionOnly(true)
+	{
+
+	}
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Worker type name"))
+	FString WorkerTypeName;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Worker permissions"))
+	FWorkerPermissionsSection WorkerPermissions;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Max connection capacity limit (0 = Unlimited capacity)", ClampMin = "0", UIMin = "0"))
+	int32 MaxConnectionCapacityLimit;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Login rate limit enabled"))
+	bool bLoginRateLimitEnabled;
+
+	UPROPERTY(EditAnywhere, config, meta = (EditCondition = "bLoginRateLimitEnabled", ConfigRestartRequired = false, DisplayName = "Login rate limit"))
+	FLoginRateLimitSection LoginRateLimit;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Rectangle grid column count", ClampMin = "1", UIMin = "1"))
+	int32 Columns;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Rectangle grid row count", ClampMin = "1", UIMin = "1"))
+	int32 Rows;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Flags"))
+	TMap<FString, FString> Flags;
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Manual Worker Connection Only"))
+	bool ManualWorkerConnectionOnly;
+};
+
+USTRUCT()
+struct FSpatialLaunchConfigDescription
+{
+	GENERATED_BODY()
+
+	FSpatialLaunchConfigDescription()
+		: Template(TEXT("small"))
+		, World()
+	{
+		FWorkerTypeLaunchSection UnrealWorkerDefaultSetting;
+		UnrealWorkerDefaultSetting.WorkerTypeName = TEXT("UnrealWorker");
+		UnrealWorkerDefaultSetting.Rows = 1;
+		UnrealWorkerDefaultSetting.Columns = 1;
+		UnrealWorkerDefaultSetting.ManualWorkerConnectionOnly = true;
+
+		Workers.Add(UnrealWorkerDefaultSetting);
+	}
+
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Template"))
```

Comment:
this is another field that may benefit from a tooltip describing possible values - or even an enum depending on how many options there are - i'm not sure how many are out there